proposals/spline-animation/README.md

```diff
@@ -0,0 +1,1980 @@
+| **Interpolation** | Flexible, typically Beziers | Linear                |
+
+## High-Level Requirements
+
+USD Anim will aim to achieve all of the following:
+
+- **Interchange-suitable:** a universal format.
+
+    - USD Anim splines will support a variety of features, detailed in
+      subsequent sections.  All clients will be able to read all valid USD Anim
+      splines, given a minimum level of feature support.  USD Anim will provide
+      a set of utilities to perform _reduction_, in which splines that use
+      features outside the client's feature set can be converted to splines
+      within the client's feature set, using emulations.  For example, if
+      clients support only Bezier segments, held and linear segments can be
+      emulated using Bezier segments.
+
+    - USD Anim will not be designed to accommodate extensions.  If the community
+      discovers missing features, we would prefer to either consider their
+      inclusion directly in the USD Anim core, or assist with the design of a
+      layered architecture that relegates features to higher levels.
+
+    - USD Anim will not include any arbitrary tuning parameters.  All behavior
+      will be mathematically justified, and all curve parameters will be
+      controllable.
+
+- **Majority Maya compatibility:** a familiar format.
+
+    - Most Maya animation splines will be representable in USD with minimal
+      translation.  In particular, USD Anim will support most Maya spline
+      features.
+
+    - However, USD Anim will be a product-agnostic format, and will differ from
+      Maya in some ways, including API, serialization format, and mathematical
+      representation.
+
+- **Majority Presto compatibility:** a pragmatic format for Pixar.
+
+    - USD Anim will be an open-source revision of technology that already exists
+      within Pixar's Presto animation system.
+
+    - Pixar's plan is to operate Presto on top of USD Anim, with an adapter
+      layer.
+
+    - This requirement should be mostly invisible to the outside world.
+      However, it may influence the decisions Pixar makes about USD Anim.
+
+- **Fast Hermite evaluation:** a runtime-friendly format.
+
+    - USD Anim will support cubic Bezier curve segments, which are probably the
+      spline model in widest use.  Bezier curves are parametric, so evaluating a
+      curve at a given time may require an iterative solving process.  Pixar has
+      found the cost of the solve negligible in the context of executing complex
+      rigging, but it may not be fast enough for all applications.
+
+    - USD Anim will also support cubic Hermite curve segments.  These differ
+      from Bezier curves in that they have one fewer degree of freedom: tangents
+      have fixed lengths, and only their slopes may be specified.  In return,
+      Hermite curves are non-parametric, and can be evaluated in constant time
+      via a simple cubic polynomial.
```

Comment:
I think it'd be helpful to define this Hermite function. It's hard to conceptualise how this explicit Hermite function works alongside the parametric Cubic Bezier function, given the same knot data.

Reply:
These Hermite curves will be as described in AnimX, c.f. https://github.com/Autodesk/animx/blob/5db8ee416d5fa7050357f498d4dcfaa6ff3f7738/src/internal/interpolators.h#L139